sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -86,6 +86,7 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
             if (msg.content.value == _TIMEOUT_MAGIC) {
                 return -ETIMEDOUT;
             }
+            __attribute__((fallthrough));
```

Comment:
There seems to be some problem using the comment in an if clause:
```
#ifdef MODULE_XTIMER
        case _TIMEOUT_MSG_TYPE:
            /* FALLTHROUGH */ 
            if (msg.content.value == _TIMEOUT_MAGIC) {
            /* FALLTHROUGH */ 
                return -ETIMEDOUT;
            /* FALLTHROUGH */ 
            }   
            /* FALLTHROUGH */ 
#endif
```
does not fix the Warning. 
Putting the `/* Falls through. */` after the `#endif` works, but doesn't seem like a good solution as the case only exists in the if clause.